lib/infer.js

```diff
@@ -1,32 +1,13 @@
 'use strict';
 
-const stageMap = [
-  {
-    stage: 'development',
-    regex: /-(dev|development)$/
-  },{
-    stage: 'test',
-    regex: /-(uat|tst|test)$/
-  },{
-    stage: 'qa',
-    regex: /-qa$/
-  },{
-    stage: 'staging',
-    regex: /-(stg|staging)$/
-  },{
-    stage: 'production',
-    regex: /-(prd|prod|production|admin|demo)$/
-  },{
-    stage: 'review',
-    regex: /-pr-(\d+)$/
-  }
-];
+const stages = require('./stages').stages;
 
 module.exports = function infer(app) {
-  for (var val of stageMap) {
-    if(val.regex.test(app)) {
-      return [app.replace(val.regex,""), val.stage];
-    }
+  const inferredStage = stages.find(stage => stage.inferRegex.test(app));
+
+  if (inferredStage) {
+    return [ app.replace(inferredStage.inferRegex, ''), inferredStage.name ];
   }
-  return [app,"production"];
+
+  return [ app, 'production' ];
```

Comment:
I'm assuming that the first element of the returned array is the stem of the app name, if a stage was found. The second element is the stage.

Would you mind refactoring this to use an object instead?